chirp/drivers/retevis_ha1g.py

```diff
@@ -0,0 +1,2111 @@
+                serial, self.read_page_len)
+            if (not flag
+               or (len(new_data_bytes) > 14 and new_data_bytes[14] != item)):
+                time.sleep(retry_delay)
+                continue
+            status.cur += len(current_page_bytes)
+            self.status_fn(status)
+            break
+
+
+def get_write_item_bytes(all_bytes: bytearray, item):
+    """
+       Get the bytes for a specific memory region
+       - Uses MEMORY_REGIONS_RANGES to find start and length
+    """
+    if item not in MEMORY_REGIONS_RANGES:
+        LOG.debug(f"Unknown memory Range:{item}")
+        return b""
+    start, length = MEMORY_REGIONS_RANGES[item]
+    return all_bytes[start:start+length]
+
+
+def read_item_handle_connect_event(all_bytes: bytearray,
+                                   item_bytes: bytes, item):
+    """
+    Put the given region bytes into the main memory array
+    - Uses MEMORY_REGIONS_RANGES to find start and length
+    """
+    if item not in MEMORY_REGIONS_RANGES:
+        LOG.debug(f"Unknown memory Range:{item}")
+    start, length = MEMORY_REGIONS_RANGES[item]
+    item_len = min(len(item_bytes), length)
+    all_bytes[start:start + item_len] = item_bytes[:item_len]
+    return all_bytes
+
+
+def exit_programming_mode(self):
+    serial = self.pipe
+    try:
+        serial.write(Get_ReadItemBytes(111, 0, 0))
+    except Exception as e:
+        LOG.debug(
+            f"Radio refused to exit programming mode:{e}")
+        raise errors.RadioError(
+            "Radio refused to exit programming mode")
+
+
+def Get_ReadItemBytes(data_type: int, page_index: int,
+                      data_code: int = 2):
+    data_part = (
+        b"RDTP\x01\x00\x00\x00\x00\x00\x00\x00\x08\x00"
+        + data_type.to_bytes(1, "little")
+        + data_code.to_bytes(1, "little")
+        + b"\x00\x00\x01\x00"
+        + page_index.to_bytes(2, "little"))
+    crc = calculate_crc16(data_part, 2)
+    return data_part + crc + b"\xff"
+
+
+def get_write_item_send_bytes(data_type: int,
+                              page_index: int,
+                              page_count: int,
+                              data_buffer: bytes):
+    data_part = (
+        b"RDTP\x01\x00\x00\x00\x00\x00\x00\x00"
+        + (len(data_buffer) + 6).to_bytes(2, "little")
+        + data_type.to_bytes(1, "little")
+        + b"\x00"
+        + page_index.to_bytes(2, "little")
+        + page_count.to_bytes(2, "little")
+        + data_buffer)
+    return data_part + calculate_crc16(data_part, 2) + b"\xff"
+
+
+def get_crc_bytes(buf, start, length):
+    cksum = 0x0000
+    for b in buf[start:start+length]:
+        cksum ^= b
+        for _ in range(8):
+            if cksum & 0x0001:
+                cksum = (cksum >> 1) ^ 0xA001
+            else:
+                cksum >>= 1
+    return struct.pack("<H", cksum)
+
+
+def calculate_crc16(byte_array, result_length) -> bytes:
+    return get_crc_bytes(byte_array, 0, len(byte_array))
```

Comment:
This function does not need to exist because it only calls the other function and the other is only used here. Please combine.